p2p/pkg/preconfirmation/tracker/tracker.go

```diff
@@ -222,14 +230,29 @@ func (t *Tracker) handleNewL1Block(
 		}
 		startTime := time.Now()
 
+		var commitmentIdx [32]byte
+		copy(commitmentIdx[:], commitment.CommitmentIndex[:])
+
+		bidAmt, ok := new(big.Int).SetString(commitment.Bid.BidAmount, 10)
```

Comment:
No point to make `bidAmt` as `*big.Int`
```suggestion
		bidAmt, err := strconv.ParseUint(commitment.Bid.BidAmount, 10, 64)
 ```

Reply:
This value will overflow because of the conversion further in the code, so converting to bigInt will not help here
`bidAmt.Uint64()`